docs/src/layouts/Main.astro

```diff
@@ -5,14 +5,18 @@ import ThemeToggle from '../components/ThemeToggle.tsx';
 import DocSidebar from '../components/DocSidebar.tsx';
 import MenuToggle from '../components/MenuToggle.tsx';
 
-const { content } = Astro.props;
+const { content = {} } = Astro.props;
 const headers = content?.astro?.headers;
-const currentPage = Astro.request.url.pathname;
-const currentFile = currentPage === '/' ? 'src/pages/index.md' : `src/pages${currentPage.replace(/\/$/, "")}.md`;
-const githubEditUrl = `https://github.com/snowpackjs/astro/blob/main/docs/${currentFile}`;
+const currentPage = Astro.request?.url?.pathname;
+let currentFile;
+let githubEditUrl;
+if (currentPage) {
```

Comment:
Does this change add any error handling? It seems like it would just end up with empty strings if there is no currentPage rather than an incorrect URL..